satpy/readers/pmw_channels_definitions.py

```diff
@@ -0,0 +1,415 @@
+            return ((self.central - self.side - self.bandwidth/2. <=
+                     other[0] - other[1] - other[2]/2. and
+                     self.central - self.side + self.bandwidth/2. >=
+                     other[0] - other[1] + other[2]/2.) or
+                    (self.central + self.side - self.bandwidth/2. <=
+                     other[0] + other[1] - other[2]/2. and
+                     self.central + self.side + self.bandwidth/2. >=
+                     other[0] + other[1] + other[2]/2.))
+
+        with suppress(AttributeError):
+            if self.unit != other.unit:
+                raise NotImplementedError("Can't compare frequency ranges with different units.")
+            return ((self.central - self.side - self.bandwidth/2. <=
+                     other.central - other.side - other.bandwidth/2. and
+                     self.central - self.side + self.bandwidth/2. >=
+                     other.central - other.side + other.bandwidth/2.) or
+                    (self.central + self.side - self.bandwidth/2. <=
+                     other.central + other.side - other.bandwidth/2. and
+                     self.central + self.side + self.bandwidth/2. >=
+                     other.central + other.side + other.bandwidth/2.))
+
+        return False
+
+    def distance(self, value):
+        """Get the distance from value."""
+        if self == value:
+            try:
+                left_side_dist = abs(value.central - value.side - (self.central - self.side))
+                right_side_dist = abs(value.central + value.side - (self.central + self.side))
+                return min(left_side_dist, right_side_dist)
+            except AttributeError:
+                if isinstance(value, (tuple, list)):
+                    return abs((value[0] - value[1]) - (self.central - self.side))
+
+                left_side_dist = abs(value - (self.central - self.side))
+                right_side_dist = abs(value - (self.central + self.side))
+                return min(left_side_dist, right_side_dist)
+        else:
+            return np.inf
+
+    @classmethod
+    def convert(cls, frq):
+        """Convert `frq` to this type if possible."""
+        if isinstance(frq, dict):
+            return cls(**frq)
+        return frq
+
+
+class FrequencyRangeBase(NamedTuple):
+    """Base class for frequency ranges.
+
+    This is needed because of this bug: https://bugs.python.org/issue41629
+    """
+
+    central: float
+    bandwidth: float
+    unit: str = "GHz"
+
+
+class FrequencyRange(FrequencyRangeBase):
+    """The Frequency range class.
+
+    The elements of the range are central and bandwidth values, and optionally
+    a unit (defaults to GHz). No clever unit conversion is done here, it's just
+    used for checking that two ranges are comparable.
+
+    This type is used for passive microwave sensors.
+
+    """
+
+    def __eq__(self, other):
+        """Return if two channel frequencies are equal.
```

Comment:
This contradicts the implementation.  It also returns true if the other is a number contained by self.